contracts/Avocado.sol

```diff
@@ -0,0 +1,128 @@
+pragma solidity ^0.4.17;
+
+contract Avocado {
+    enum PersonType { Teacher, Student }
+
+    // Person Structure
+    struct Person {
+        // Teacher or Student
+        PersonType personType;        
+        string name;        
+        string description;                        
+        uint ethPerHour;
+
+        // Array of the Hash of the active meetings
+        bytes32[] activeMeetings;
+
+        // Array of the Hash of completed meetings
+        bytes32[] completedMeetings;
+                
+        bool exists;
+    }
+
+    // Meeting structure
+    struct Meetings {
+        address student;
+        address teacher;
+        string description;
+        uint32 timeInEpoch; // When the session's gonna happen
+        uint8 meetingDuration; // How long is the meeting going to be        
+    }
+
+    // Reviews
+    struct Ratings {
+        address addr;
+        uint rating; // out of 5
+        string description; // Short review description
+    }
+
+    // Hash of Meeting -> IsTeacher (Based on UserType) -> Review Object
+    mapping(bytes32 => mapping(bool => Ratings[])) public ratings;
```

Comment:
This is pretty cool, but I'm not sure I understand what the `bool` is there for? I understand that this is a mapping of a mapping (i.e. nested dictionary), but I'm not sure why the inner mapping is keyed by a `bool`.

Reply:
I've been informed by @joshma91 that the `bool` is how you check for whether or not the person is a teacher or a student. For the sake of readability, can we not type alias this?

Reply:
Ethereum only supports using primitive types as keys unfortunately

Reply:
😞 sad